docs/development/testguide.rst

```diff
@@ -684,44 +684,23 @@ Testing warnings
 ================
 
 In order to test that warnings are triggered as expected in certain
-situations, you can use the `astropy.tests.helper.catch_warnings`
-context manager.  Unlike the `warnings.catch_warnings` context manager
-in the standard library, this one will reset all warning state before
-hand so one is assured to get the warnings reported, regardless of
-what errors or warnings may have been emitted by other tests previously.
-Here is a real-world example::
-
-  from astropy.tests.helper import catch_warnings
-
-  with catch_warnings(MergeConflictWarning) as warning_lines:
-      # Test code which triggers a MergeConflictWarning
-      out = table.vstack([t1, t2, t4], join_type='outer')
-
-      assert warning_lines[0].category == metadata.MergeConflictWarning
-      assert ("In merged column 'a' the 'units' attribute does not match (cm != m)"
-              in str(warning_lines[0].message))
-
```

Comment:
Should the example be replaced with an analogous one for `pytest.warns`, perhaps demonstrating the `match` option and how to return the warning? `pytest.raises` has its own example, and it would save one from following an external link for a quick overview.

Reply:
> Should the example be replaced with an analogous one

I debated with myself on this one... We can barely keep up with doc updates as it is, adding example that you can find on official site anyway seems like we're just giving ourselves more work for no reason. If there is strong opinion of adding an example back, I'll do it.

Reply:
I think on balance I agree with @pllim here. Overall, the recommendation might be to generally follow what is in the test files already. But not in this PR.... Might be worth a separate issue to look over the testguide more generally.